### PR TITLE
DP-1380: expose influxDB behind same authentication as monitoring

### PR DESCRIPTION
**Pull request by AlexFernandes-MOVAI**
DP-1380: expose influxDB behind same authentication as monitoring (EE 2.3.1)

### Content
- update of Documentation with latest additions
- align all configuration files of develop / qa / release environment
- 

### Test influxdb API access:
- build and tag the docker image
- use it in a running robot
- `curl "https://admin:admin123@localhost/influxdb/query?q=SHOW+
DATABASES" -k`

### CODE REVIEW

File: config/dev/haproxy_develop.cfg [hunk 1]
```diff
@@ -81,12 +81,24 @@ frontend ha-frontend-http
...
+    # InfluxDB authenticated ACL
+    acl PATH_influxdb path_beg -i /influxdb
+    acl is_authusers_acl http_auth(AuthUsers)
+    http-request auth realm ha-backend-influxdb if PATH_influxdb !is_authusers_acl
+    use_backend ha-backend-influxdb if { path -i -m beg /influxdb }
+
+    # Ros Tools rewriting ACL
+    acl url_ros-tools path_beg -i /ros-tools
+    acl url_ros-tools_websockify path_beg -i /websockify
+    use_backend ha-backend-ros-tools if url_ros-tools or url_ros-tools_websockify
```
Comment (DenisBorzenkov) [inline config/dev/haproxy_develop.cfg:101]: ok

File: config/release/haproxy_release.cfg [hunk 2]
```diff
@@ -89,6 +89,12 @@ frontend ha-frontend-http
     http-request auth realm ha-backend-monitoring if PATH_monitoring !is_authusers_acl
     use_backend ha-backend-monitoring if { path -i -m beg /monitoring }
 
+    # InfluxDB authenticated ACL
+    acl PATH_influxdb path_beg -i /influxdb
+    acl is_authusers_acl http_auth(AuthUsers)
+    http-request auth realm ha-backend-influxdb if PATH_influxdb !is_authusers_acl
+    use_backend ha-backend-influxdb if { path -i -m beg /influxdb }
+
```
Comment (DenisBorzenkov) [inline config/release/haproxy_release.cfg:97]: ok

**Review by DenisBorzenkov — COMMENTED** [review]
ok

File: README.md [hunk 3]
```diff
@@ -37,6 +38,24 @@
...
+- To add more users dynamically to this group, the `haproxy.cfg` file can be updated with the new user's credentials.
+  Passwords should be hashed using the `mkpasswd` command as follows:
+
+```bash
+mkpasswd -m sha-256 <password>
+```
+
+- To remove a user from the group, the `haproxy.cfg` file can be updated to remove the user.
+- The `haproxy.cfg` file should be updated with the new user's credentials and the container should be restarted to apply the changes.
+
```
Comment (DenisBorzenkov) [inline README.md:58]: ok

**Review by DenisBorzenkov — APPROVED** [review]

**Comment by mariana-dias-alves** [comment]
> curl "https://admin:admin123@localhost/influxdb/query?q=SHOW+ DATABASES" -k

@diasdm I tried with this version:
`$ curl -k "https://a
dmin:admin123@localhost/influxdb/query?q=SHOW%20DATABASES"`

**Review by mariana-dias-alves — APPROVED** [review]

**Comment by AlexFernandes-MOVAI** [comment]
> After setup, for `curl "https://admin:admin123@localhost/influxdb/query?q=SHOW+ DATABASES" -k` I get:
> 
> ```
> curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
> ```
> 
> And it seems the ros-tools url requires a trailing slash: `https://localhost/ros-tools/`

Yes, see Mariana's answer. Curl can have different handling of spaces in URLs, it is better to "urlify" it

**Comment by AlexFernandes-MOVAI** [comment]
> After setup, for `curl "https://admin:admin123@localhost/influxdb/query?q=SHOW+ DATABASES" -k` I get:
> 
> ```
> curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
> ```
> 
> And it seems the ros-tools url requires a trailing slash: `https://localhost/ros-tools/`

yes, I thin this was always the case. Same for `/monitoring` and `/influxdb` but I can double chek

**Review by diasdm — APPROVED** [review]

**Comment by diasdm** [comment]
> yes, I thin this was always the case. Same for `/monitoring` and `/influxdb` but I can double chek

Its just a detail, but I can actually reach [https://localhost/monitoring](https://localhost/monitoring), [https://localhost/monitoring/](https://localhost/monitoring/) and [https://localhost/ros-tools/](https://localhost/ros-tools/), but not [https://localhost/ros-tools](https://localhost/ros-tools)

**Pull request merged**